### PR TITLE
Implement asset registry

### PR DESCRIPTION
**Pull request by benxgao**
Populate asset selection list with on-chain data rather than hard-coded local variables.

### CODE REVIEW

**Comment by benxgao** [comment]
@KarishmaBothara  Here is draft ui change that may help with the investigation in rpc implement, thanks.

**Comment by benxgao** [comment]
Test select-dropdown working, sending tx is not broken.

Kapture 2020-07-17 at 10 07 35

File: packages/app-generic-asset/src/Transfer.tsx [hunk 1]
```diff
@@ -45,16 +44,27 @@ function Transfer ({ assets, className, onClose, recipientId: propRecipientId, s
   let user = senderId;
   const assetBalance = useCall<Balance>(api.query.genericAsset.freeBalance as any, [id, user]);
 
-  // build up our list of options via assets
+  const retrivedAssetList = useCall<any>(
```
Comment (holygits) [inline packages/app-generic-asset/src/Transfer.tsx:47]: nit: `retrievedAssets`

**Comment by lilian79511** [comment]
Ben brought out one concern that to change from getting data from hard-coded local variables to the on-chain data, the feature to save user's behavior on the browser is removed. Hence the asset id stay unselected every time they open the GA modal.

From design's point of view it's fine to keep it unselected for now.

File: packages/app-generic-asset/src/Transfer.tsx (hunk 1 above)
Comment (holygits) [reply packages/app-generic-asset/src/Transfer.tsx:47]: @benxgao this is spelt wrong would've been an easy fix before merge

**Pull request merged**